### PR TITLE
Explain and link to proof specs. 

### PR DESCRIPTION
**Pull request by msporny**
Related to #470.

/cc @jricher

***
<a href="https://pr-preview.s3.amazonaws.com/w3c/vc-data-model/pull/593.html" title="Last updated on May 7, 2019, 1:50 AM UTC (4816ada)">Preview</a> | <a href="https://pr-preview.s3.amazonaws.com/w3c/vc-data-model/593/bb5ef42...4816ada.html" title="Last updated on May 7, 2019, 1:50 AM UTC (4816ada)">Diff</a>

### CODE REVIEW

File: index.html [hunk 1]
```diff
@@ -988,6 +989,17 @@ <h3>Concrete Lifecycle Example</h3>
...
         </pre>
+
+        <p class="note">
+Implementers that are interested in understanding more about the
+<code>proof</code> mechanism used above can learn more in
+Section <a href="#proofs-signatures"></a> and by reading the
+following specifications: Linked Data Proofs [[LD-PROOFS]], Linked Data
+Signatures [[LD-SIGNATURES]], 2018 RSA Signature Suite [[LDS-RSA2018]], and
+JSON Web Signature (JWS) Unencoded Payload Option [[RFC7797]]. A current list
+of available proof mechanisms is available in the Verifiable Credentials
```
Comment (brentzundel) [inline index.html]: recommend changing to:
"A list of proof mechanisms is available in the Verifiable Credentials . . ."

File: index.html [hunk 2]
```diff
@@ -1571,7 +1583,14 @@ <h3>Proofs (Signatures)</h3>
...
         <p class="note">
 No single proof mechanism is currently standardized or recommended for
-<a>verifiable credentials</a>.
+<a>verifiable credentials</a>. Implementers that are interested in
+understanding more about the <code>proof</code> mechanism can
+learn more by reading the following specifications:
+Linked Data Proofs [[LD-PROOFS]], Linked Data Signatures [[LD-SIGNATURES]],
+2018 RSA Signature Suite [[LDS-RSA2018]], and JSON Web Signature (JWS)
+Unencoded Payload Option [[RFC7797]]. A current list of available proof
+mechanisms is available in the Verifiable Credentials Extension Registry
```
Comment (brentzundel) [inline index.html]: same as above, recommend changing to:
"A list of proof mechanisms is available in the Verifiable Credentials . . ."

File: index.html (hunk 1 above)
Comment (msporny) [reply index.html]: Fixed in 4816ada7ed318fa2f13888569581376a0f71804d.

File: index.html (hunk 2 above)
Comment (msporny) [reply index.html]: Fixed in 4816ada7ed318fa2f13888569581376a0f71804d.

**Pull request merged**